project-keeper/src/test/java/com/exasol/projectkeeper/validators/OwnVersionValidatorTest.java

```diff
@@ -0,0 +1,108 @@
+
+import java.util.List;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import com.exasol.projectkeeper.mavenrepo.MavenRepository;
+import com.exasol.projectkeeper.validators.OwnVersionValidator.Updater;
+import com.exasol.projectkeeper.validators.finding.SimpleValidationFinding;
+import com.exasol.projectkeeper.validators.finding.ValidationFinding;
+
+//[utest->dsn~verify-own-version~1]
+//[itest->dsn~verify-own-version~1]
+//[utest->dsn~self-update~1]
+//[itest->dsn~self-update~1]
+class OwnVersionValidatorTest {
+
+    private static String expectedMessage(final int number, final String suffix) {
+        return "W-PK-CORE-" + number + ": Could not detect latest .* " + suffix;
+    }
+
+    private static final String OUTDATED = "W-PK-CORE-153: Project-keeper version 0.0.1 is outdated."
+            + " Please update project-keeper to latest version.*";
+
+    @Test
+    void retrieveLatestVersion_IoError() {
+        verifyOptionalFinding(testee("0.1.0"), expectedMessage(155, "Connection refused.*"));
+    }
+
+    @Test
+    void ownVersionUnsupportedFormat() {
+        verifyOptionalFinding(testee("a.b.c"), //
+                "W-PK-CORE-152: Could not validate version of project-keeper."
+                        + " Unsupported format of own version.*");
+    }
+
+    @Test
+    void latestVersionUnsupportedFormat() throws Exception {
+        verifyOptionalFinding(testee("0.0.1", "x.y.z"),
+                expectedMessage(154, "Unsupported format of latest version from Maven repository.*"));
+    }
+
+    @Test
+    void ownVersionOutDatet_ProposeUpdate() throws Exception {
+        verifyOptionalFinding(testee("0.0.1", "99999999"), OUTDATED);
+    }
+
+    @Tag("integration")
+    @Test
+    void versionUpToDate_NoFinding() throws Exception {
+        final String currentVersion = MavenRepository.mavenPlugin().getLatestVersion();
+        final OwnVersionValidator testee = OwnVersionValidator //
+                .forMavenPlugin(currentVersion, mock(Updater.class));
+        assertThat(testee.validate(), empty());
+    }
+
+    @Tag("integration")
+    @Test
+    void cli_NoFinding() throws Exception {
+        final String currentVersion = MavenRepository.cli().getLatestVersion();
+        final OwnVersionValidator testee = OwnVersionValidator.forCli(currentVersion);
+        assertThat(testee.validate(), empty());
+    }
+
+    @Tag("integration")
+    @Test
+    void cli_NoFix() throws Exception {
+        final OwnVersionValidator testee = OwnVersionValidator.forCli("0.0.1");
+        final SimpleValidationFinding finding = (SimpleValidationFinding) testee.validate().get(0);
+        assertThat(finding.getMessage(), matchesRegex(OUTDATED));
+        assertThat(finding.hasFix(), is(false));
+    }
+
+    // private helpers
+
+    private OwnVersionValidator testee(final String currentVersion) {
+        return testee(currentVersion, new MavenRepository("https://localhost/unknown/end/point"));
+    }
```

Comment:
- Rename method to "testeeWithInvalidMavenRepo".
- Check if argument is necessary or could be null